agent/xds/rbac.go

```diff
@@ -566,6 +590,68 @@ func makeRBACRules(
 	return rbac
 }
 
+func collectJWTClaims(rbacIxn *rbacIntention) []*structs.IntentionJWTClaimVerification {
+	var claims []*structs.IntentionJWTClaimVerification
+
+	if len(rbacIxn.Claims) > 0 {
+		claims = append(claims, rbacIxn.Claims...)
+	}
+
+	for _, perm := range rbacIxn.Permissions {
```

Comment:
I wonder if the `claims` from the each permissions should be handle differently. These could be path specific but it doesn't seem to me like i can specify a specific http path in the metadata

example: 

```

Kind = "service-intentions"
Name = "redis"

JWT = {
  Providers = [
    {
      // Provider must contain the name of an existing jwt-provider.
      Name = "okta"

      // VerifyClaims contains the path and value for claims to verify
      // beyond what is configured by the jwt-provider.
      VerifyClaims = [
        {
            Path = ["aud"]
            Value = "api.apps.organization.com"
        }
      ]
    }
  ]
}

Sources = [
  {
    Name = "*"
    Permissions = {
      Action = "allow"
      HTTP = {
        PathPrefix = "/v2"
        Methods    = ["GET"]
      }
      JWT = {
        Providers = [
          {
            // Provider must contain the name of an existing jwt-provider.
            Name = "okta"

            // VerifyClaims contains the path and value for claims to verify
            // beyond what is configured by the jwt-provider.
            VerifyClaims = [
              {
                  Path = ["aud"]
                  Value = "api.apps.organization.com"
              },
              {
                  Path = ["perm", "role"]
                  Value = "create.organization.com"
              }
            ]
          },
        ]
      }

    }
  }
]
```

will end up something along the lines of: 

```
name: envoy.filters.http.jwt_authn
        config:
          providers:
            okta:
              issuer: ....
              payload_in_metadata: jwt_payload
          rules:
          - match: {prefix: /v2}
            requires: {provider_name: okta}
      - name: envoy.filters.http.rbac
        config:
          rules:
            action: ALLOW
            policies:
                principals:
                   - metadata:
                       filter: envoy.filters.http.jwt_authn
                       path:
                         - key: jwt_payload
                         - key: aud
                       value:
                         string_match:
                           exact: "api.apps.organization.com"
                   - metadata:
                       filter: envoy.filters.http.jwt_authn
                       path:
                         - key: jwt_payload
                         - key: perm
                         - role
                       value:
                         string_match:
                           exact: "create.organization.com"

```

this seems fine to me but would like to know your thoughts: @pglass